service/service_test.go

```diff
@@ -0,0 +1,78 @@
+package service
+
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"os"
+	"testing"
+	"time"
+)
+
+func createMockService() *Service {
+
+	return &Service{
+		Marshaller:    MockMarshaller{},
+		InitialOffset: int64(-1),
+		Schema:        getDefaultSchema(),
+	}
+}
+
+func TestUnitStart(t *testing.T) {
+	Convey("Successful process of a single Kafka message", t, func() {
+
+		c := make(chan os.Signal)
+		svc := createMockService()
+
+		Convey("Given a message is readily available for the service to consume", func() {
+			svc.Consumer = createMockConsumerWithMessage()
+
+			Convey("When the upsert request is successful", func() {
+
+				svc.Upsert = MockUpsert{}
+
+				Convey("Then the consumer will close succesfully", func() {
```

Comment:
The `Then` clause should be asserting correct behaviour only. This is a bit misleading as the service doesn't terminate the consumer after a message processed.